### DISCUSSION TITLE
version control with google cloud shell

### DISCUSSION THREAD

**Discussion by its-dibo** [coder/code-server#2780] [General]
I installed code-server in Google cloud shell.
it installed successfully and opened, but I cannot use the version control.

from Google shell I run `git version` and it returned the value git version 2.20.1, whitch means that git is already installed.

also I installed `git extension pack`, but it didn't help.

I run `git status` inside the project and got the following:
```
On branch master
Your branch is up to date with 'origin/master'
```

![image](https://user-images.githubusercontent.com/57308531/109213328-221e5380-77b9-11eb-842c-148d6e1ad549.png)

**Comment by code-asher** [comment]
The message here suggests you haven't opened a git repository yet. You can do that with `File > Open Folder`. If you don't have one yet you can create one with `git init`.

**Comment by its-dibo** [comment]
Thanks, @code-asher for your response,
This message appears even if I opened a folder (project initiated with git) or if I cloned a repo from github.

**Reply by code-asher** [reply]
Ahhh interesting. You've probably already done this but can you double-check that the file explorer shows the root of the project?

Another thing is to test is to check if VS Code can see `git`. Are you able to run `git` commands on the project when opening the bulit-in terminal?

**Reply by its-dibo** [reply]
yes, the file explorer shows the project's root and yes I can run git commands from the VS built-in terminal.

**Reply by code-asher** [reply]
Strange! I haven't had any luck replicating yet (using 3.9.0). There might be some information in the logs that will help pinpoint the issue: https://github.com/cdr/code-server/blob/v3.9.0/docs/FAQ.md#how-do-i-debug-issues-with-code-server

**Reply by its-dibo** [reply]
it seems that `--log debug` did a magic, for no reason the issue solved!

**Reply by code-asher** [reply]
Hahaha magical! Maybe code-server just needed to be restarted for some reason. :thinking:

**Comment by its-dibo** [comment]
The problem occurred again even when passing `--log debug`
the command `git status` works!

I think the issue is that code-server couldn't activate the extensions 

![image](https://user-images.githubusercontent.com/57308531/109995140-b129ef00-7d16-11eb-94ab-0e42d7b5daae.png)


log:

![image](https://user-images.githubusercontent.com/57308531/110005320-01a64a00-7d21-11eb-9168-1eb701a62ba1.png)


I see that extensions don't displayed in the status bar (ex: prettier), also "accounts" just continuously loading....

I restarted the cloud shell and opened it in a new clean Google chrome instance without any browser-extension enabled.

**Reply by code-asher** [reply]
Ahh yeah that does seem like it could cause the issue. Could you post the browser console log?

**Comment by its-dibo** [comment]
@code-asher 
The issue **may** be caused due to a network connection problem, preventing the extensions to be loaded, in addition to preventing the connection between code-server and the remote repo to be established .
so I suggest in case of a connection failure, `code-server` must display an obvious warning message.

**Reply by code-asher** [reply]
Ahh that does make sense. I think it only shows a message if it fails to connect and not when the connection is interrupted so we'll probably want some logic to handle that.

**Comment by bpmct** [comment]
So I'm curious how you're using code-server with Google Cloud Shell. I believe cloud shell will reset the system after you disconnect? Are you re-installing it every time?
